Unithereum.CodeGen/Config.cs

```diff
@@ -0,0 +1,279 @@
+                            {
+                                var cat = CharUnicodeInfo.GetUnicodeCategory(c);
+                                return cat == UnicodeCategory.UppercaseLetter ||
+                                       cat == UnicodeCategory.LowercaseLetter ||
+                                       cat == UnicodeCategory.TitlecaseLetter ||
+                                       cat == UnicodeCategory.ModifierLetter ||
+                                       cat == UnicodeCategory.OtherLetter ||
+                                       cat == UnicodeCategory.LetterNumber ||
+                                       cat == UnicodeCategory.SpacingCombiningMark ||
+                                       cat == UnicodeCategory.DecimalDigitNumber ||
+                                       cat == UnicodeCategory.ConnectorPunctuation ||
+                                       c == '.'
+                                    ? c.ToString()
+                                    : "_";
+                            }))));
+
+            var firstLetterCat = CharUnicodeInfo.GetUnicodeCategory(result[0]);
+            if (
+                firstLetterCat != UnicodeCategory.UppercaseLetter &&
+                firstLetterCat != UnicodeCategory.LowercaseLetter &&
+                firstLetterCat != UnicodeCategory.TitlecaseLetter &&
+                firstLetterCat != UnicodeCategory.ModifierLetter &&
+                firstLetterCat != UnicodeCategory.OtherLetter &&
+                firstLetterCat != UnicodeCategory.LetterNumber)
+                result = "_" + result;
+
+            result = string.Join(
+                ".",
+                result
+                    .Split(".")
+                    .Where(part => !string.IsNullOrWhiteSpace(part))
+                    .Select(part => CsharpKeywords.Contains(part) ? "@" + part : part));
+
+            return result;
+        }
+
+        private static string SanitizeLeadingDots(string str)
+        {
+            return str != string.Empty ? str[0] == '.' ? "_" + SanitizeLeadingDots(str[1..]) : str : string.Empty;
+        }
+
+        private static string SanitizeTrailingDots(string str)
+        {
+            return str != string.Empty ? str[^1] == '.' ? SanitizeTrailingDots(str[..^1]) + "_" : str : string.Empty;
+        }
+    }
+}
+
+public class ConfigDeserializer : JsonConverter
+{
+    public override object? ReadJson(JsonReader reader, Type objectType, object? existingValue,
+        JsonSerializer serializer)
+    {
+        if (reader.TokenType == JsonToken.Null)
+            return null;
+
+        var target = new JObject();
+
+        foreach (var jToken in JToken.Load(reader).Children())
+        {
+            var property = (JProperty)jToken;
+            if (char.IsUpper(property.Name[0]))
+            {
+                Debug.LogWarning($"Unithereum CodeGen: invalid config property {property.Name}. "
+                                 + "Use camelCase in codegen.config.json.");
```

Comment:
```suggestion
                                 + "Hint: valid keys for codegen.config.json are in camelCase.");
```

Might be more friendly.

Also, I think it might be unnecessary to have a special case only for PascalCase. We might be able to get away with displaying the hint whenever an unrecognized key is encountered.

Reply:
I agree this case is unnecessary. Just removing case for PascalCase would be better.